src/mlpack/tests/recurrent_network_test.cpp

```diff
@@ -1413,4 +1414,82 @@ BOOST_AUTO_TEST_CASE(BRNNTrainReturnObjective)
   BOOST_REQUIRE_EQUAL(std::isfinite(objVal), true);
 }
 
+/**
+ * Test a BRNN model with PrintLoss callback and optimizer.
+ */
+BOOST_AUTO_TEST_CASE(BRNNWithOptimizerCallbackTest)
+{
+  const size_t rho = 10;
+
+  arma::cube input;
+  arma::mat labelsTemp;
+  GenerateNoisySines(input, labelsTemp, rho, 6);
+
+  arma::cube labels = arma::zeros<arma::cube>(1, labelsTemp.n_cols, rho);
+  for (size_t i = 0; i < labelsTemp.n_cols; ++i)
+  {
+    const int value = arma::as_scalar(arma::find(
+        arma::max(labelsTemp.col(i)) == labelsTemp.col(i), 1)) + 1;
+    labels.tube(0, i).fill(value);
+  }
+
+  Add<> add(4);
+  Linear<> lookup(1, 4);
+  SigmoidLayer<> sigmoidLayer;
+  Linear<> linear(4, 4);
+  Recurrent<>* recurrent = new Recurrent<>(
+      add, lookup, linear, sigmoidLayer, rho);
+
+  BRNN<> model(rho);
+  model.Add<IdentityLayer<> >();
+  model.Add(recurrent);
+  model.Add<Linear<> >(4, 5);
+  std::stringstream stream;
+  
+  StandardSGD opt(0.1, 1, 500 * input.n_cols, -100);
+  model.Train(input, labels, opt, PrintLoss(stream));
+  BOOST_TEST_CHECKPOINT("Training over");
+
+  BOOST_REQUIRE_GT(stream.str().length(), 0);
+}
+
+/**
+ * Test a BRNN model with PrintLoss callbacks.
+ */
+BOOST_AUTO_TEST_CASE(BRNNCallbackTest)
+{
+  const size_t rho = 10;
+
+  arma::cube input;
```

Comment:
Strange, looks like the test failed for the same model; I think you used the same model for each callback test?

Reply:
Yes , is there any issue , since Between each boost test case destructor are called !

Reply:
Actually I'm not sure, let's see if I can replicate the issue on the Mac setup I have.

Reply:
@zoq any update on this , were you able to replicate ?

Reply:
Sorry, haven't had a chance yet to take a look, will put it on my list for tomorrow.

Reply:
@zoq is there anything missing in this pull request, can we help??

Reply:
If you like you can try to reproduce the error in ther macOS build: https://travis-ci.org/mlpack/mlpack/jobs/594423997?utm_medium=notification&utm_source=github_status